app/src/main/java/de/grobox/transportr/trips/search/DirectionsViewModel.kt

```diff
@@ -0,0 +1,209 @@
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.grobox.transportr.trips.search
+
+import android.util.Pair
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import de.grobox.transportr.TransportrApplication
+import de.grobox.transportr.data.locations.FavoriteLocation.FavLocationType
+import de.grobox.transportr.data.locations.LocationRepository
+import de.grobox.transportr.data.searches.SearchesRepository
+import de.grobox.transportr.favorites.trips.SavedSearchesViewModel
+import de.grobox.transportr.locations.LocationLiveData
+import de.grobox.transportr.locations.LocationView.LocationViewListener
+import de.grobox.transportr.locations.WrapLocation
+import de.grobox.transportr.networks.TransportNetworkManager
+import de.grobox.transportr.networks.getTransportNetwork
+import de.grobox.transportr.settings.SettingsManager
+import de.grobox.transportr.trips.TripQuery
+import de.grobox.transportr.trips.search.TripsRepository.QueryMoreState
+import de.grobox.transportr.ui.TimeDateFragment.TimeDateListener
+import de.grobox.transportr.utils.DateUtils
+import de.grobox.transportr.utils.LiveTrigger
+import de.grobox.transportr.utils.SingleLiveEvent
+import de.schildbach.pte.NetworkId
+import de.schildbach.pte.dto.Product
+import de.schildbach.pte.dto.Trip
+import java.util.*
+import javax.annotation.ParametersAreNonnullByDefault
+import javax.inject.Inject
+
+@ParametersAreNonnullByDefault
+class DirectionsViewModel @Inject internal constructor(
+    application: TransportrApplication, transportNetworkManager: TransportNetworkManager, settingsManager: SettingsManager,
+    locationRepository: LocationRepository, searchesRepository: SearchesRepository
+) : SavedSearchesViewModel(application, transportNetworkManager, locationRepository, searchesRepository), TimeDateListener, LocationViewListener {
+
+    private val _tripsRepository: TripsRepository
+    private val _fromLocation = MutableLiveData<WrapLocation?>()
+    private val _viaLocation = MutableLiveData<WrapLocation?>()
+    private val _toLocation = MutableLiveData<WrapLocation?>()
+    val locationLiveData: LocationLiveData
+    val findGpsLocation = MutableLiveData<FavLocationType?>()
+    val timeUpdate = LiveTrigger()
+    private val _now = MutableLiveData<Boolean>()
+    private val _calendar = MutableLiveData<Calendar?>()
+    private val _products = MutableLiveData<EnumSet<Product>>()
+    private val _isDeparture = MutableLiveData<Boolean>()
+    private val _isExpanded = MutableLiveData<Boolean>()
+    val showTrips = SingleLiveEvent<Void>()
+    @JvmField
+    val topSwipeEnabled = MutableLiveData<Boolean>()
```

Comment:
why does this need a JvmField annotation?

Reply:
Good question, I guess the IntelliJ converter added this. Will recheck.

Reply:
Removed.